kura/org.eclipse.kura.protocol.can.test/src/main/java/org/eclipse/kura/protocol/can/test/CanSocketTest.java

```diff
@@ -108,42 +99,35 @@ public void setup() {
 
     @Test
     public void startCanTestThread() {
-        if (this.m_pollThread != null) {
-            this.m_pollThread.interrupt();
-            try {
-                this.m_pollThread.join(100);
-            } catch (InterruptedException e) {
-                // Ignore
-            }
-            this.m_pollThread = null;
-        }
+        terminatePollThread();
 
-        this.m_pollThread = new Thread(new Runnable() {
+        this.pollThread = new Thread(new Runnable() {
 
             @Override
             public void run() {
-                if (CanSocketTest.this.m_canConnection != null && !IS_MAC) {
-                    while (!CanSocketTest.this.thread_done) {
-                        CanSocketTest.this.thread_done = doCanTest();
+                if (CanSocketTest.this.canConnection != null && !IS_MAC) {
+                    boolean threadDone = false;
+                    while (!threadDone) {
+                        threadDone = doCanTest();
                     }
                 }
             }
         });
-        this.m_pollThread.start();
+        this.pollThread.start();
     }
 
     public boolean doCanTest() {
         byte[] b;
         CanMessage cm = null;
-        if (this.m_isMaster) {
-            if (this.m_orig >= 0) {
+        if (this.isMaster) {
```

Comment:
I can agree with you about `this.interfaceName`, `this.canId ` and `this.isMaster` that are variables that, in theory, can be affected by multithreading.
The others, in my opinion, are effectively managed by a single thread and do not need to be volatile.